### PR TITLE
Support for different comment types

### LINKED ISSUE #45: Add support for function documentation
Add:

## Example:
```elm
{-| my function that is awesome -}
function : Awesome
function = Awesome
```
->
```elixir
@doc "my function that is awesome"
@spec function :: :awesome
def function(), do: :awesome
```

This is an AST job because elm-ast sees no difference between block and regular comments

### PR DESCRIPTION
**Pull request by Baransu**
Fixes #45

### CODE REVIEW

File: src/ExStatement.elm [hunk 1]
```diff
@@ -111,6 +129,36 @@ elixirS c s =
             notImplemented "statement" s
 
 
+indentComment : Context -> String -> String
+indentComment { indent } content =
+    content
+        |> prependAll (List.repeat ((indent + 1) * 2) " " |> String.join "")
```
Comment (wende) [inline src/ExStatement.elm]: Can be replaced with `indAll`

File: src/ExStatement.elm [hunk 2]
```diff
@@ -111,6 +129,36 @@ elixirS c s =
...
+        |> String.dropRight 1
```
Comment (wende) [inline src/ExStatement.elm]: What's the purpose of that?

File: src/ExStatement.elm [hunk 3]
```diff
@@ -111,6 +129,36 @@ elixirS c s =
...
+
+
+getCommentType : String -> ElmchemyComment
+getCommentType comment =
+    [ ( "^\\sex", (\s -> Ex s) )
```
Comment (wende) [inline src/ExStatement.elm]: What would kids say?

File: src/Helpers.elm [hunk 4]
```diff
@@ -12,6 +13,16 @@ type MaybeUpper
...
+indent : Context -> Context
+indent c =
+    { c | indent = c.indent + 1 }
+
+
+deindent : Context -> Context
+deindent c =
+    { c | indent = c.indent - 1 }
+
+
```
Comment (wende) [inline src/Helpers.elm]: I think that's more suitable in ExContext than here.
Helpers.elm file is meant to be left agnostic against the rest of the project files

File: src/ExStatement.elm (hunk 2 above)
Comment (Baransu) [inline src/ExStatement.elm]: Because of something there is unnecessary `\n` at the end.

File: src/Helpers.elm (hunk 4 above)
Comment (Baransu) [inline src/Helpers.elm]: Right

File: src/ExStatement.elm (hunk 3 above)
Comment (Baransu) [inline src/ExStatement.elm]: https://github.com/wende/scalixir

**Pull request merged**